contract/src/mechanics/claim.rs

```diff
@@ -384,6 +429,87 @@ mod tests {
+            vesting_period,
+            instant_claim_percentage: None,
+            vesting_scheme: VestingScheme::AfterCliff,
+        });
+        let investment = InvestmentAmount {
+            amount: 80_000,
+            weight: 80_000_000,
+            claimed: 0,
+        };
+        let total_sold_tokens = 92_000_000;
+
+        // Exactly before cliff period ends
+        let cliff_end_timestamp = config.end_date + cliff_period;
+        let claim_res =
+            available_for_claim(&investment, total_sold_tokens, &config, cliff_end_timestamp)
+                .unwrap();
+        assert_eq!(claim_res, 0);
+
+        // Halfway through vesting period
+        let current_timestamp = cliff_end_timestamp + vesting_period.as_nanos() / 2;
+        let res = available_for_claim(&investment, total_sold_tokens, &config, current_timestamp)
+            .unwrap();
+        let expected_calc = (investment.weight * config.sale_amount.0 / total_sold_tokens)
+            * (u128::from(current_timestamp - cliff_end_timestamp))
+            / u128::from(vesting_period.as_nanos());
+        assert_eq!(res, expected_calc);
+    }
+
+    #[test]
+    fn test_vesting_scheme_after_cliff_with_instant_claim() {
+        let mut config = price_discovery_config();
+        config.sale_amount = 200_000_000_000_000_000.into();
+        let vesting_period = 2_000_000.into();
+        let cliff_period = 500_000;
+        config.vesting_schedule = Some(VestingSchedule {
+            cliff_period: cliff_period.into(),
+            vesting_period,
+            instant_claim_percentage: Some(1_100), // 11%
+            vesting_scheme: VestingScheme::AfterCliff,
+        });
+        let investment = InvestmentAmount {
+            amount: 80_000,
+            weight: 80_000_000,
+            claimed: 0,
+        };
+        let total_sold_tokens = 92_000_000;
+        let assets_for_claim = 80_000_000 * config.sale_amount.0 / total_sold_tokens;
+
+        let instant_claim_res =
+            available_for_claim(&investment, total_sold_tokens, &config, config.end_date).unwrap();
+        let expected_calc_instant_claim = assets_for_claim * 11 / 100;
+        assert_eq!(instant_claim_res, expected_calc_instant_claim);
+
+        let cliff_end_timestamp = config.end_date + cliff_period;
+        let instant_claim_res = available_for_claim(
+            &investment,
+            total_sold_tokens,
+            &config,
+            cliff_end_timestamp - 1,
+        )
+        .unwrap();
+        assert_eq!(instant_claim_res, expected_calc_instant_claim);
+
+        // Halfway through vesting period
+        let current_timestamp = cliff_end_timestamp + vesting_period.as_nanos() / 2;
+        let res = available_for_claim(&investment, total_sold_tokens, &config, current_timestamp)
+            .unwrap();
+        let expected_calc = (investment.weight * config.sale_amount.0 / total_sold_tokens)
+            * (u128::from(current_timestamp - config.end_date))
```

Comment:
Here should be:

```suggestion
            * (u128::from(current_timestamp - cliff_end_timestamp))
```

Because the vesting scheme is `AfterCliff`.